servers/rendering/rasterizer_rd/shaders/tonemap.glsl

```diff
@@ -251,7 +251,11 @@ vec3 apply_bcs(vec3 color, vec3 bcs) {
 }
 
 vec3 apply_color_correction(vec3 color, sampler3D correction_tex) {
-	return texture(correction_tex, color).rgb;
+	ivec3 tex_size = textureSize(correction_tex, 0);
+	color = texture(correction_tex,
+			color * (vec3(tex_size - ivec3(1)) / vec3(tex_size)) + vec3(0.5) / vec3(tex_size))
+					.rgb;
+	return color;
```

Comment:
That's not very readable. Maybe instead:
```suggestion
	vec3 coord = color * (vec3(tex_size - ivec3(1)) / vec3(tex_size)) + vec3(0.5) / vec3(tex_size);
	return texture(correction_text, coord).rgb;
```

And further for the actual equation:
```
	vec3 coord = (color * vec3(tex_size - ivec3(1)) + vec3(0.5)) / vec3(tex_size);
```
Should also be faster.

Reply:
This equation doesn't work. I had trouble and wanted someone else to look over this part.

Reply:
The color correction is still broken, but updated code.